### PR TITLE
[CDAP-8259] Documentation for dataset permissions and reuse properties

### PR DESCRIPTION
**Pull request by anew**

### CODE REVIEW

**Comment by anew** [comment]
Docs Quick Build running here: http://builds.cask.co/browse/CDAP-DQB289-1

File: cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst [hunk 1]
```diff
@@ -84,9 +85,15 @@ or deleting of files: it treats the contents of the base path as read-only::
...
 
-If you do not specify an input format, you will not be able to use this as the input for a
-MapReduce program; similarly for the output format.
+If you want to use an existing location and still be able to write to it, you have two options:
 
+- ``setUseExisting(true)``: This directs the FileSet to accept an existing location as its base
+  path. However, because the existing location may contain files prior to the FileSet creation,
+  the location will not be deleted when the dataset is dropped, and truncating the FileSet will
+  have no effect. This is to ensure that no pre-existing data is deleted.
+- ``setPossessExisting(true)``: Similarly, this allows to reuse an existing location. In addition,
```
Comment (awholegunch) [inline cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst]: What about:
```
Similarly, this allows reuse of an existing location. The FileSet will assume ownership of existing files
in that location, which means that these files will be deleted when the dataset is dropped or truncated.

**Comment by awholegunch** [comment]
Trying out Quick Build: http://builds.cask.co/browse/CDAP-DQB289-4

Pages of interest: 
- http://builds.cask.co/artifact/CDAP-DQB289/shared/build-4/Docs-HTML/html/developers-manual/building-blocks/datasets/fileset.html#creating-a-fileset
- http://builds.cask.co/artifact/CDAP-DQB289/shared/build-4/Docs-HTML/html/developers-manual/building-blocks/datasets/permissions.html
- http://builds.cask.co/artifact/CDAP-DQB289/shared/build-4/Docs-HTML/html/developers-manual/data-exploration/filesets.html#limitations

LGTM

**Comment by anew** [comment]
Thanks for the updates, @awholegunch, they look good. I will squash commits and merge.

**Pull request merged**